### PR TITLE
OSDOCS-12851: Documented the 4.15.40 z-stream release notes

### PR DESCRIPTION
**Pull request by dfitzmau**
Version(s):
4.15

Issue:
[OSDOCS-12851](https://issues.redhat.com/browse/OSDOCS-12851)

Link to docs preview:
* [4.15.40](https://85919--ocpdocs-pr.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-15-release-notes.html#ocp-4-15-40_release-notes)

### CODE REVIEW

File: release_notes/ocp-4-15-release-notes.adoc [hunk 1]
```diff
@@ -2775,6 +2775,39 @@ This section will continue to be updated over time to provide notes on enhanceme
...
+
+* Previously, {hcp}-based clusters were unable to authenticate through the `oc login` command. The web browser displayed an error when it attepted to retrieve the the token after selecting *Display Token*. with this release, `cloud.ibm.com` and other cloud-based endpoints are no longer proxied and authentication is successful. (link:https://issues.redhat.com/browse/OCPBUGS-44278[*OCPBUGS-44278*])
+
+* Previously, the Messaging Application Programming Interface (MAPI) for {ibm-cloud-title} currently only checks the first group of subnets (50) when searching for subnet details by name. With this release, the search provides pagination support to search all subnets. (link:https://issues.redhat.com/browse/OCPBUGS-43675[*OCPBUGS-43675*])
```
Comment (ShaunaDiaz) [inline release_notes/ocp-4-15-release-notes.adoc:2804]:
```suggestion
* Previously, the Messaging Application Programming Interface (MAPI) for {ibm-cloud-title} only checked the first group of subnets (50) when searching for subnet details by name. With this release, the search provides pagination support to search all subnets. (link:https://issues.redhat.com/browse/OCPBUGS-43675[*OCPBUGS-43675*])
```

File: release_notes/ocp-4-15-release-notes.adoc [hunk 2]
```diff
@@ -2775,6 +2775,39 @@ This section will continue to be updated over time to provide notes on enhanceme
...
+
+[source,terminal]
+----
+$ oc adm release info 4.15.40 --pullspecs
+----
+
+[id="ocp-4-15-40-bug-fixes_{context}"]
+==== Bug fixes
+
+* Previously, the Single-Root I/O Virtualization (SR-IOV) Operator did not expire the acquired lease during the Operator's shutdown operation. This impacted a new instance of the Operator, because the new instance had to wait for the lease to expire before the new instance was operational. With this release, an update to the Operator shutdown logic ensures that the Operator expires the lease when the Operator is shutting down. (link:https://issues.redhat.com/browse/OCPBUGS-43361[*OCPBUGS-43361*])
```
[the remark this answers is not in the document]
Comment (dfitzmau) [reply release_notes/ocp-4-15-release-notes.adoc]: :heavy_check_mark:

File: release_notes/ocp-4-15-release-notes.adoc [hunk 3]
```diff
@@ -2775,6 +2775,39 @@ This section will continue to be updated over time to provide notes on enhanceme
...
+
+* Previously, when you used the Agent-based Installer to install a cluster on a node that had an incorrect date, the cluster installation failed. With this release, a patch is applied to the Agent-based Installer live ISO time synchronization. The patch configures the `/etc/chrony.conf` file with the list of additional Network Time Protocol (NTP) servers, so that you can set any of these additional NTP servers in the `agent-config.yaml` without experiencing a cluster installation issue. (link:https://issues.redhat.com/browse/OCPBUGS-45207[*OCPBUGS-45207*])
```
[the remark this answers is not in the document]
Comment (dfitzmau) [reply release_notes/ocp-4-15-release-notes.adoc]: Added "the patch fixes the date issue".

File: release_notes/ocp-4-15-release-notes.adoc [hunk 4]
```diff
@@ -2775,6 +2775,39 @@ This section will continue to be updated over time to provide notes on enhanceme
...
+
+* Previously, {hcp}-based clusters were unable to authenticate through the `oc login` command. The web browser displayed an error when it attepted to retrieve the the token after selecting *Display Token*. with this release, `cloud.ibm.com` and other cloud-based endpoints are no longer proxied and authentication is successful. (link:https://issues.redhat.com/browse/OCPBUGS-44278[*OCPBUGS-44278*])
```
[the remark this answers is not in the document]
Comment (dfitzmau) [reply release_notes/ocp-4-15-release-notes.adoc]: :heavy_check_mark:

File: release_notes/ocp-4-15-release-notes.adoc (hunk 1 above)
Comment (dfitzmau) [reply release_notes/ocp-4-15-release-notes.adoc]: :heavy_check_mark:

**Pull request merged**